book_maker/translator/chatgptapi_translator.py

```
@@ -136,25 +217,77 @@ def create_context_messages(self):
+        t_text = ""
+        max_len_retry= 2
+        for len_retry in range(max_len_retry):
+            cur_content = completion.choices[0].message.content
+            cur_content = re.sub(r'<think>.*?</think>','',cur_content,flags=re.S) #r1-like things
+            if cur_content is not None:
+                t_text += cur_content.encode("utf8").decode() or ""
+            else:
+                break
+            if completion.choices[0].finish_reason != "length":
+                break
+            if needprint:
```

Ruff (0.15.12)

[error] Undefined name `needprint`

(F821)